pkg/reference/reference.go

```diff
@@ -213,32 +239,33 @@ func (r *APIResolver) Resolve(ctx context.Context, req ResolutionRequest) (Resol
 		return ResolutionResponse{ResolvedValue: req.CurrentValue, ResolvedReference: req.Reference}, nil
 	}
 
-	// The reference is already set - resolve it.
-	if req.Reference != nil {
-		if err := r.client.Get(ctx, types.NamespacedName{Name: req.Reference.Name}, req.To.Managed); err != nil {
-			return ResolutionResponse{}, errors.Wrap(err, errGetManaged)
+	// The selector is set - resolve it.
+	if req.Selector != nil {
```

Comment:
This results in behavior change for cases where both ref and selector are given and no policy is set. Chatted offline about possible solutions.

Reply:
As we discussed, I preserved the order (first Reference field then Selector field). 

For new Always policy scenario, I implemented a possible solution (we discussed offline) that sets the Reference field to nil when Selector is not empty and resolve policy is always.